benchmarks/src/test/java/io/crate/data/join/RowsBatchIteratorBenchmark.java

```diff
@@ -74,58 +72,6 @@
     private final List<Row1> oneThousandRows = IntStream.range(0, 1000).mapToObj(Row1::new).collect(Collectors.toList());
     private final List<Row1> tenThousandRows = IntStream.range(0, 10000).mapToObj(Row1::new).collect(Collectors.toList());
 
-    @Benchmark
-    public void measureConsumeBatchIterator(Blackhole blackhole) throws Exception {
-        BatchIterator<Row> it = new InMemoryBatchIterator<>(rows, SENTINEL);
-        while (it.moveNext()) {
-            blackhole.consume(it.currentElement().get(0));
-        }
-    }
-
-    @Benchmark
-    public void measureConsumeCloseAssertingIterator(Blackhole blackhole) throws Exception {
-        BatchIterator<Row> it = new InMemoryBatchIterator<>(rows, SENTINEL);
-        BatchIterator<Row> itCloseAsserting = new CloseAssertingBatchIterator<>(it);
-        while (itCloseAsserting.moveNext()) {
-            blackhole.consume(itCloseAsserting.currentElement().get(0));
-        }
-    }
-
-    @Benchmark
-    public void measureConsumeSkippingBatchIterator(Blackhole blackhole) throws Exception {
-        BatchIterator<Row> it = new InMemoryBatchIterator<>(rows, SENTINEL);
-        BatchIterator<Row> skippingIt = new SkippingBatchIterator<>(it, 100);
-        while (skippingIt.moveNext()) {
-            blackhole.consume(skippingIt.currentElement().get(0));
-        }
-    }
-
-    @Benchmark
-    public void measureConsumeNestedLoopJoin(Blackhole blackhole) throws Exception {
-        BatchIterator<Row> crossJoin = JoinBatchIterators.crossJoin(
-            InMemoryBatchIterator.of(oneThousandRows, SENTINEL),
-            InMemoryBatchIterator.of(tenThousandRows, SENTINEL),
-            new CombinedRow(1, 1)
-        );
-        while (crossJoin.moveNext()) {
-            blackhole.consume(crossJoin.currentElement().get(0));
-        }
-    }
-
-    @Benchmark
-    public void measureConsumeNestedLoopLeftJoin(Blackhole blackhole) throws Exception {
-        BatchIterator<Row> leftJoin = JoinBatchIterators.leftJoin(
-            InMemoryBatchIterator.of(oneThousandRows, SENTINEL),
-            InMemoryBatchIterator.of(tenThousandRows, SENTINEL),
-            new CombinedRow(1, 1),
-            row -> Objects.equals(row.get(0), row.get(1))
-        );
-        while (leftJoin.moveNext()) {
-            blackhole.consume(leftJoin.currentElement().get(0));
-        }
-        leftJoin.moveToStart();
-    }
-
     @Benchmark
     public void measureConsumeHashInnerJoin(Blackhole blackhole) {
         BatchIterator<Row> leftJoin = JoinBatchIterators.hashInnerJoin(
```

Comment:
should we create the iterator once before the execution? (since we are moving it to start at the end?) 

otherwise, we shouldn't be resetting the iterator as part of the benchmark?

Reply:
hm I didn't change those stuff so the iterator is always created on each benchmark iteration. The moveToStart() is unecessary I think. right?